### PR TITLE
Update Dockerfile to base on osgeo/gdal instead of geobase, update constraints and requirements to increment versions

### PR DESCRIPTION
**Pull request by alexgleith**
TL;DR simplify Dockerfile by using a base that has geo libraries installed in it and increment odc libs.

The geobase image is a complicated build environment that is difficult to understand when builds are failing.

The main change in this PR is to switch to `osgeo/gdal`, which is well maintained image that is put together by [someone much smarter than me](https://github.com/OSGeo/gdal/tree/master/gdal/docker) and I think it's a good base install of those key geospatial libraries, while being small and simple.

The changes in this PR will make it easier to increment versions of key libraries.

### CODE REVIEW

**Comment by alexgleith** [comment]
Thoughts now, @omad?
